File: CSharp/Project/Talk/BadFruit.cs
#region The Worst of Legacy Code: Forensic Development
#pragma warning disable 0649, 0642
#endregion

using System.Collections.Generic;
using Talk.Required;

namespace Talk
{
    public class BadFruit : Fruit
    {
        private Person _user;

        public void CreateLoans(params Loan[] кредитов)
        {
            ОбщиеКонфигурация.Видыпервоначальногокредита();
            System.Data.Common.DbConnection подключение = new ОбщиеКонфигурация().получитьконфигурациюбазыданных().GetDatabaseConnectionFor(this._user);
            List<Person> люди = new List<Person>();
            List<int?> индексыинвалидов = new List<int?>();
            // создать массив для всех людей, вовлеченных в виде кредитов
            foreach (Loan кредит in кредитов)
            {
                люди.AddRange(кредит.GetPeopleOnLoan());
            }

            // найти дубликаты всех индексов
            if (индексыинвалидов.Count < 25)
            {
                for (int я = 1; я < люди.Count; я++)
                {
                    if (люди.GetRange(0, я - 1).Contains(люди[я]))
                    {
                        индексыинвалидов.Add(я);
                    }
                }
            }
            else
            {
                кредитов = Получатьвсезаймыдлятекущегопользователя();
                foreach (Loan кредит in кредитов)
                {
                    люди.AddRange(кредит.GetPeopleOnLoan());
                }

            }
            for (int я = 0; я < люди.Count; я++)
            {
                if (!индексыинвалидов.Contains(я)) ;
                {
                    Save(люди[я], подключение);
                }
                Увеличениенагрузкинаграфа(люди[я]);
            }
        }

        // Bug #54
        //If a home , auto and personal loan are taken out by tom,
        //3 versions of tom appear in the database.
    }
}

File: EcmaScript/program/app/badFruit.js
'use strict';

const красныйЛента = require('../app/ignored/заставляйтеменязарабатыватьденьгиналечение');

// The Worst of Legacy Code: Forensic Development

function createLoans() 
{
    красныйЛента();
    const информационнаясистема = require('./ignored/можетбытьтолькоодин'); 
    let денежныесредства = [];
    let ещераз = [];

    let финансовыйвклад = Array.prototype.slice.call(arguments);
    // создать массив для всех людей, вовлеченных в виде кредитов
    денежныесредства = (
        финансовыйвклад
        .reduce((acc, Деньги) => acc.concat(Деньги.getBorrowers()), денежныесредства)
    );
    // найти дубликаты всех индексов
    if(ещераз.length < 25) {
        for (let я = 1; я < денежныесредства.length; я++) {
            if(денежныесредства.slice(0, я - 1).includes(денежныесредства[я])) {
                ещераз.push(я);
            }
        }
    }
    else {
        финансовыйвклад = красныйЛента(ещераз, 'важныйслучай');
        денежныесредства = (
            финансовыйвклад
            .reduce((кочевник, Деньги) => кочевник.concat(Деньги.getBorrowers()), денежныесредства)
        );
    }
    for (let я = 0; я < денежныесредства.length; я++) {
        if(!ещераз.includes(я)) ;
        {
            информационнаясистема.add(денежныесредства[я]);
        }
        красныйЛента(денежныесредства[я]);
    }
}

// Bug #54
//If a home , auto and personal loan are taken out by tom,
//3 versions of tom appear in the database.

const getMoneyFromPeople = {
    createLoans: createLoans
};

module.exports = getMoneyFromPeople;